lib/osnetwork.rb

```diff
@@ -4,55 +4,84 @@
 require_relative 'osproject_android'
 
 class NetworkHandler
-    @instance = new
+  @instance = new
 
-    private_class_method :new
+  private_class_method :new
 
-    def self.instance
-      @instance
-    end
+  URL = 'https://api.onesignal.com/api/v1/track'
 
-    def track_uri
-      URI.parse('https://api.onesignal.com/api/v1/track')
-    end
+  def self.instance
+    @instance
+  end
 
-    def getHttp_net(uri)
-      http = Net::HTTP.new(uri.host, uri.port)
-      http.use_ssl = true
-      return http
-    end
+  def get_http_net()
+    uri = URI.parse(URL)
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.use_ssl = true
+    return http
+  end
+
+  def send_track_from_error(app_id, platform, lang, success_mesage, error_message)
+    error_message = "error=#{error_message}"
+    actions_taken = ""
```

Comment:
I think you don't need this definition, as you will define it in both cases in the `if`.